cmd/homi/setup/cmd.go

```diff
@@ -786,6 +810,66 @@ func makeSCNodes(num int, isWorkOnSingleHost bool) ([]*ValidatorInfo, []string)
 	return scn, scnNodeKeys
 }
 
+func makeSPNNodes(num int, isWorkOnSingleHost bool) ([]*ValidatorInfo, []string) {
+	privKeys, nodeKeys, nodeAddrs := istcommon.GenerateKeys(num)
+
+	var p2pPort uint16
+	var proxies []*ValidatorInfo
+	var proxyNodeKeys []string
+	for i := 0; i < num; i++ {
+		if isWorkOnSingleHost {
+			p2pPort = lastIssuedPortNum
+			lastIssuedPortNum++
```

Comment:
lastIssuedPortNum is a global variable. Is there a shared resource issue here?

Reply:
There is no concurrency issue here.